### PR TITLE
Add error handling for updating division

### LINKED ISSUE #199: Show state results when updating division
This is currently missing the following:
* Showing the user that it is being handled
* Showing the user if it has failed
* Disable interaction if it is being handled
* Tests

Criteria: 
- [ ] The user should know that their name is being added when they click on it
- [ ] The user should not be able to interact with that instrument when it is still updating the division
- [ ] Also implement error handling for this
- [ ] Tests should be written for these edge cases

### PR DESCRIPTION
**Pull request by KevinBreurken**
Closes #199 

Adds error handling for when the division is inserted or deleted. During that process, the selected Instrument is in its loading state, and after updating division and retrieving the updated song from back-end, it leaves loading state.

I tried to make the updating of the divisions asynchronous, but due to the way the page retrieves a new song after sending a request, it resulted in weird behaviour making it difficult to implement.

### CODE REVIEW

**Comment by fkoulen** [comment]
> I tried to make the updating of the divisions asynchronous, but due to the way the page retrieves a new song after sending a request, it resulted in weird behaviour making it difficult to implement.

Asynchronous in what way?

File: src/components/utils/spinner-stripes.tsx [hunk 1]
```diff
@@ -2,12 +2,13 @@ import React from "react"
 
 interface SpinnerStripesProps {
   dataCy?: string
+  tailwindProps?: string
 }
 
-const SpinnerStripes = ({ dataCy }: SpinnerStripesProps) => {
+const SpinnerStripes = ({ dataCy, tailwindProps = "stroke-white" }: SpinnerStripesProps) => {
```
Comment (fkoulen) [inline src/components/utils/spinner-stripes.tsx:8]: Totally forgot that you can set a default value like that! Gonna quickly do this in my branch 🤫

**Comment by KevinBreurken** [comment]
> > I tried to make the updating of the divisions asynchronous, but due to the way the page retrieves a new song after sending a request, it resulted in weird behaviour making it difficult to implement.
> 
> Asynchronous in what way?

When you press multiple instruments quickly after eachother, so that the user sends insert/delete divisions for multiple instruments. The process right now sends the division request, and after that the front-end sends another request to retrieve the song again. The issue that i faced was that it would update the instrumentsInUpdate incorrectly due to useState related issues being asynchronous-like.

I tried to look for a queueing method, but i didn't get the result that felt favourable and was quite hacky implemented. The implementation right now on this branch prevents multiple requests by disabling the other instruments. It's not that noticeable on a normal connection, but you can check how it works with networking set to Fast/3G

File: src/components/images/song-preview-image.tsx
Comment (rensdoodeman) [inline src/components/images/song-preview-image.tsx:1]: why are you reformatting this? is it you IDE doing this or you preference?

File: src/pages/suggestions/[suggestion].tsx
Comment (rensdoodeman) [inline src/pages/suggestions/[suggestion].tsx:1]: You could look at debouncing for the async effect you're looking for, for an example: I've done it for e.g. the suggestion-information-area. Would be something like `debounce(() => ..your code, 400)`

File: src/components/images/song-preview-image.tsx
Comment (KevinBreurken) [reply src/components/images/song-preview-image.tsx:1]: Yep, prettier runs on commit, So merge conficts also formats added files.

File: src/pages/suggestions/[suggestion].tsx
Comment (KevinBreurken) [reply src/pages/suggestions/[suggestion].tsx:1]: That would require to combine the insert/deletes and queue those requests. It's too late to implement that functionality into this PR.

**Comment by fkoulen** [comment]
> > > I tried to make the updating of the divisions asynchronous, but due to the way the page retrieves a new song after sending a request, it resulted in weird behaviour making it difficult to implement.
> 
> > 
> 
> > Asynchronous in what way?
> 
> 
> 
> When you press multiple instruments quickly after eachother, so that the user sends insert/delete divisions for multiple instruments. The process right now sends the division request, and after that the front-end sends another request to retrieve the song again. The issue that i faced was that it would update the instrumentsInUpdate incorrectly due to useState related issues being asynchronous-like.
> 
> 
> 
> I tried to look for a queueing method, but i didn't get the result that felt favourable and was quite hacky implemented. The implementation right now on this branch prevents multiple requests by disabling the other instruments. It's not that noticeable on a normal connection, but you can check how it works with networking set to Fast/3G

It might be too late now, but the attending members list is updated in realtime by subscribing to it.
